enforcer-rules/README.md

```diff
@@ -3,6 +3,31 @@
 This Maven enforcer rule verifies that the transitive dependency tree of `pom.xml` does not have
 any [linkage errors](../library-best-practices/glossary.md#types-of-conflicts-and-compatibility).
 
+This rule performs the following steps:
+
+1. Build a dependency tree of Maven artifacts with the project as the root node
+
+   In the tree, a Maven artifact has directed edges to other Maven artifacts when the artifact's
+   pom.xml declares dependencies to them. The root node of the tree is the Maven artifact
+   of the project in which the enforce rule runs.
+
+   **`Optional` and `provided`**: Unlike Maven's resolution algorithm, the tree includes transitive
+   dependencies marked as `optional` or `provided`.
+   This inclusion helps to detect incompatibilities beyond Maven's build class path.
+1. Build a class path from the dependency tree
+
+   It traverses the tree in level-order to generate a list of Maven artifacts.
+1. Verify the references in the class path (list of JAR files)
+
+   It performs the following steps:
+   - From all of the class files in the input JAR files, it reads class, method and field
```

Comment:
Does it read the whole class file, including the body of all the methods?
It's not clear to me how it reads all these references? Using reflection?

Reply:
It's from constant pool section of a class file.

Continuing review at https://docs.google.com/document/d/1fD6eCj-xfR9Ef3iudYRuxKFPeB51akrNUgWiV7gR6Gc/edit?usp=sharing